plugins/keysend.c

```diff
@@ -13,26 +14,58 @@ static void init(struct plugin *p, const char *buf UNUSED,
 static const struct plugin_command commands[] = {
 };
 
-static struct command_result *htlc_accepted_continue(struct command *cmd)
+static struct command_result *
+htlc_accepted_continue(struct command *cmd, struct tlv_tlv_payload *payload)
 {
 	struct json_stream *response;
+	u8 *binpayload, *rawpayload;
 	response = jsonrpc_stream_success(cmd);
+
 	json_add_string(response, "result", "continue");
+	if (payload) {
+		binpayload = tal_arr(cmd, u8, 0);
+		towire_tlvstream_raw(&binpayload, payload->fields);
+		rawpayload = tal_arr(cmd, u8, 0);
+		towire_bigsize(&rawpayload, tal_bytelen(binpayload));
+		towire(&rawpayload, binpayload, tal_bytelen(binpayload));
+		json_add_string(response, "payload", tal_hex(cmd, rawpayload));
```

Comment:
It seems a bit weird to prepend the length here, doesn't it?  Seems like we're exposing a (bad) internal detail?

Ah, never mind, this allows us to put in legacy payloads instead I guess.  Should we remove this ability though, since it's going to be v. ugly when legacy gets removed?

Reply:
Either works for me, legacy payloads need to die alright :+1: